synapse/client/nodes/stream_out.py

```diff
@@ -16,93 +14,101 @@
     SynapseData,
 )
 
+DEFAULT_STREAM_OUT_PORT = 50038
+
+
+# Try to get the current user's ip for setting the destination address
+def get_client_ip():
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    try:
+        # This won't actually establish a connection, but helps us figure out the ip
+        s.connect(("8.8.8.8", 80))
+        local_ip = s.getsockname()[0]
+    except Exception as e:
+        logging.error(f"Failed to get client IP: {e}")
+        return None
+    finally:
+        s.close()
+    return local_ip
+
 
 class StreamOut(Node):
     type = NodeType.kStreamOut
 
-    def __init__(self, label=None, multicast_group=None):
+    def __init__(self, label=None, destination_address=None, destination_port=None):
         self.__socket = None
         self.__label = label
-        self.__multicast_group: Optional[str] = multicast_group
 
-    def read(self) -> Optional[SynapseData]:
+        # If we have been passed a None for destination address, try to resolve it
+        if not destination_address:
+            self.__destination_address = get_client_ip()
+        else:
+            self.__destination_address = destination_address
+
+        if not destination_port:
+            self.__destination_port = DEFAULT_STREAM_OUT_PORT
+        else:
+            self.__destination_port = destination_port
+
+    def read(self) -> Tuple[Optional[SynapseData], int]:
```

Comment:
While you're here, technically this should be 
```suggestion
    def read(self) -> Tuple[Optional[Tuple[NDTPHeader, SynapseData]], int]:
```
(`_unpack`'s signature is wrong as well)

Reply:
Thank you.